design/cli-install-changes.md

```diff
@@ -0,0 +1,156 @@
+# Proposal for a more intuitive CLI to install and configure Velero
+
+Currently, the Velero CLI tool has a `install` command that configures numerous major and minor aspects of Velero. As a result, the combined set of flags for this `install` command makes it hard to intuit and reason about the different Velero components. This document proposes changes to improve the UX for installation and configuration in a way that would make it easier for the user to discover what needs to be configured by looking at what is available in the CLI rather then having to rely heavily on our documentation for the usage. At the same time, it is expected that the documentation update to reflect these changes will also make the documentation flow easier to follow.
+
+This proposal prioritizes discoverability and self-documentation over minimalizing length or number of commands and flags.
+
+## Goals
```

Comment:
I think we need to consider impacts on GitOps users as well - right now `velero install` allows users to generate a complete set of YAML that can be checked into git and applied as the user sees fit; breaking things up into multiple commands likely makes it more difficult/not possible for those users to get a complete set of YAML without applying it to the cluster.  I would probably add a goal around that.

Reply:
Here's some docs on kubectl's integration with kustomize - https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/

Reply:
What I envision for GitOps here: `velero init` would produce a base YAML that users would deploy, and the plugins would be layers that users could deploy with `kubectl apply -k` that customizes the base deployment with the appropriate init containers, secrets, etc.

Reply:
Added to TODO list.